src/main/java/domain/Result.java

```diff
@@ -1,46 +1,50 @@
 package domain;
 
+import domain.participant.BetAmount;
 import domain.participant.Hands;
 import java.util.Arrays;
 import java.util.function.BiPredicate;
+import java.util.function.Function;
 
 public enum Result {
 
-    WIN("승", Result::winningCondition),
-    TIE("무", Result::tieCondition),
-    LOSE("패", Result::loseCondition);
+    WIN(Result::winningCondition, BetAmount::keep),
+    WIN_BLACKJACK(Result::winningBlackJackCondition, BetAmount::multiply),
+    TIE(Result::tieCondition, BetAmount::zero),
+    LOSE(Result::loseCondition, BetAmount::lose);
 
-    private final String value;
     private final BiPredicate<Hands, Hands> condition;
+    private final Function<BetAmount, Profit> profitFunction;
 
-    Result(final String value, final BiPredicate<Hands, Hands> condition) {
-        this.value = value;
+    Result(BiPredicate<Hands, Hands> condition, Function<BetAmount, Profit> profitFunction) {
         this.condition = condition;
+        this.profitFunction = profitFunction;
     }
 
-    public Result reverse() {
-        if (Result.WIN.equals(this)) {
-            return LOSE;
-        }
-
-        if (Result.LOSE.equals(this)) {
-            return WIN;
-        }
-
-        return TIE;
+    public static Result calculateProfit(Hands hands, Hands target) {
```

Comment:
```suggestion
    public static Result calculateProfit(final Hands hands, final Hands target) {
```
모든 곳에 잊지 않고 final 처리하기 쉽지 않죠? 😂
intelliJ 설정을 변경해서 메서드 분리시 파라미터에 final 을 자동으로 넣어줄 수도 있긴해요!

Reply:
으앗.. 메서드 분리를 따로 안하고 바로 작성하다보니 자동완성이 안 됐네요 🥹 좀 더 신경쓰겠습니다 !!